src/main/java/org/dcsa/bkg/service/impl/BKGServiceImpl.java

```diff
@@ -119,23 +123,26 @@ public Mono<BookingResponseTO> createBooking(final BookingTO bookingRequest) {
                                 bookingTO.setVesselIMONumber(v.getVesselIMONumber());
                                 return Mono.just(bookingTO);
                               }),
-                      optionalInMono(locationService.createLocationByTO(
-                          bookingRequest.getInvoicePayableAt(),
-                          invPayAT ->
-                              bookingRepository.setInvoicePayableAtFor(invPayAT, bookingID))),
-                      optionalInMono(locationService.createLocationByTO(
-                          bookingRequest.getPlaceOfIssue(),
-                          placeOfIss ->
-                              bookingRepository.setPlaceOfIssueIDFor(placeOfIss, bookingID))),
+                      optionalInMono(
```

Comment:
Although not fully in scope of this story, consider refactoring Mono<Optional<List< to Mono<List< to adhere to new coding practices.